skimage/_shared/utils.py

```diff
@@ -547,6 +680,59 @@ def wrapped(*args, **kwargs):
         return wrapped
 
 
+def _deprecate_estimate(func, class_name=None):
+    """Deprecate ``estimate`` method."""
+    class_name = func.__qualname__.split('.')[0] if class_name is None else class_name
+    return deprecate_func(
+        deprecated_version="0.26",
+        removed_version="2.2",
+        hint=f"Please use `{class_name}.from_estimate` class constructor instead.",
+        stacklevel=2,
+    )(func)
+
+
+def _deprecate_inherited_estimate(cls):
+    """Deprecate inherited ``estimate`` instance method.
+
+    This needs a class decorator so we can correctly specify the class of the
+    `from_estimate` class method in the deprecation message.
+    """
+
+    def estimate(self, *args, **kwargs):
+        return self._estimate(*args, **kwargs) is None
+
+    # The inherited method will always be wrapped by deprecator.
+    inherited_meth = getattr(cls, 'estimate').__wrapped__
+    estimate.__doc__ = inherited_meth.__doc__
+    estimate.__signature__ = inspect.signature(inherited_meth)
+
+    cls.estimate = _deprecate_estimate(estimate, cls.__name__)
+    return cls
+
+
+def _fix_inherited_from_estimate(cls):
```

Comment:
Not sure I understand what's happening here, but trust that it's necessary :)

Reply:
I'm afraid so - I did go through various iterations on this.   Can unpack if you're interested.

Reply:
I'll assume I can resolve this issue.

Reply:
Is this just to get the name to appear correctly? If we can just expand "fix" to be a bit more explicit, that'd be enough.

Reply:
Done.